tools/sotn_str/sotn_str.py

```diff
@@ -30,17 +30,15 @@ def process_string(match: re.Match[str]):
 
 def process_string_jp(match: re.Match[str]):
     s = match.group(1)
+    s = s.replace('\\"', '"')
     out = utf8_to_byte_literals(s)
     escaped = "".join([f"\\x{val:02X}" for val in out])
     return f'"{escaped}"'
 
 
 def do_sub(line):
-    pattern = r'_S\("([^"]*)"\)'
-    # english_str_processed = re.sub(pattern, process_string, line)
-    # pattern_jp = r'_SJ\("([^"]+)"\)'
+    pattern = r'_S\("((?:\\["]|[^"])*)"\)'
     jp_str_processed = re.sub(pattern, process_string_jp, line)
-    jp_str_processed = jp_str_processed.replace("_S(\"\")", "\"\\xFF\"")
```

Comment:
Long term this might be better written as a state machine to better handle escape sequences. This now allows `\"` (escaped double quotes) in strings (on line 40). These need to be dealt with after the match (on line 33).